### PR TITLE
Coupler CI

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
Adds a second job to test compilation of the coupler with PRs.

This will fail until https://github.com/NOAA-GFDL/FMScoupler/pull/75 goes in.

**How Has This Been Tested?**
Tested on my forks

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by rem1776** [comment]
I'm gonna simplify this and just add the coupler build test to the FMS CI and run on PRs after the normal testing suite.

**Comment by thomas-robinson** [comment]
We can add model tests on parallel works too if that would help, but those
also require more resources.

Tom Robinson
Pronouns: he/him/his
Physical Scientist in the Modeling Systems Division
NOAA Geophysical Fluid Dynamics Lab
Forrestal Campus, Princeton University
201 Forrestal Road
Princeton, NJ 08540
Email: ***@***.***

**Comment by thomas-robinson** [comment]
I don't think this worked like you hoped it would.  FMS got cloned/checked out.  Then coupler was cloned and then: 
```
Run t/null_model_build.sh
  t/null_model_build.sh
  shell: sh -e {0}
Cloning into 'mkmf'...
Cloning into '/__w/FMS/FMS/t/null.O0meQo/src/FMS'...
```
It looks like it got it's own clone of FMS main
https://github.com/NOAA-GFDL/FMScoupler/blob/c12af876d4baef20346db8391422b6b6df209c75/t/null_model_build.sh#L26-L27

**Comment by rem1776** [comment]
@thomas-robinson I'll put another PR in for the coupler have the test script take a SHA to checkout as an optional argument. I'll probably try to see if I can get it actually running as well instead of just compiling.

**Review by underwoo — DISMISSED** [review]

File: .github/workflows/coupler.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on: [pull_request]
+
+jobs:
+  coupler-build:
+    runs-on: ubuntu-latest
+    container:
+      image: ryanmulhall/hpc-me.ubuntu-minimal:coupler
+      env:
+        CC: mpicc
+        FC: mpif90
```
Comment (thomas-robinson) [inline .github/workflows/coupler.yml:11]: You have the HPC-ME container, but you aren't using mpiicc and mpiifort.  Is this on purpose?

Comment (rem1776) [reply .github/workflows/coupler.yml:11]: Yeah this is with a gnu based image

**Pull request merged**